tsdb/engine/tsm1/reader.go

```diff
@@ -249,54 +249,59 @@ func (t *TSMReader) KeyAt(idx int) (string, byte) {
 
 func (t *TSMReader) ReadAt(entry *IndexEntry, vals []Value) ([]Value, error) {
 	t.mu.RLock()
-	defer t.mu.RUnlock()
-
-	return t.accessor.readBlock(entry, vals)
+	v, err := t.accessor.readBlock(entry, vals)
+	t.mu.RUnlock()
+	return v, err
 }
```

Comment:
Does the caller of this method recover? If so we could potentially have a permanently locked reader if `readBlock()` ever panicked.